### PR TITLE
fix empty http request body read panic

### PR DESCRIPTION
**Pull request by suchen-sci**
During test of my new write filter, i find if we set empty http request body `http.NewRequest(http.MethodPost, "127.0.0.1", nil)` and read body from `HTTPContext`, callbackreader will panic. this pr aims to fix this issue.

### CODE REVIEW

**Review by samutamm — APPROVED** [review]
LGTM

File: pkg/util/callbackreader/callbackreader.go [hunk 1]
```diff
@@ -51,6 +51,10 @@ func New(r io.Reader) *CallbackReader {
 }
 
 func (cr *CallbackReader) Read(p []byte) (int, error) {
+	if cr.reader == nil {
```
[the remark this answers is not in the document]
Comment (suchen-sci) [reply pkg/util/callbackreader/callbackreader.go]: good idea!

[the remark this answers is not in the document]
Comment (suchen-sci) [reply pkg/util/callbackreader/callbackreader.go]: it should be `if cr.reader != nil {n, err = cr.reader.Read(p)}`.....

[the remark this answers is not in the document]
Comment (localvar) [reply pkg/util/callbackreader/callbackreader.go]: yes, my typo

**Review by localvar — APPROVED** [review]

**Pull request merged**